File: BellDriver.py
import qsharp

from qsharp import Result
from BellEntanglement import TestBellState
# from BellEntanglement import TestBellStateBase, TestBellStateInverse, TestBellStateRandom, TestBellState

initials = (Result.Zero, Result.One)

# for i in initials:
# 	result = TestBellStateBase.simulate(count = 1000, initial = i)
# 	(num_zeros, num_ones) = result
# 	print(f"Init: {i: <4} zeros = {num_zeros: <4} ones = {num_ones: <4}")


# for i in initials:
# 	result = TestBellStateInverse.simulate(count = 1000, initial = i)
# 	(num_zeros, num_ones) = result
# 	print(f"Init: {i: <4} zeros = {num_zeros: <4} ones = {num_ones: <4}")

# for i in initials:
# 	result = TestBellStateRandom.simulate(count = 1000, initial = i)
# 	(num_zeros, num_ones) = result
# 	print(f"Init: {i: <4} zeros = {num_zeros: <4} ones = {num_ones: <4}")

for i in initials:
	result = TestBellState.simulate(count = 1000, initial = i)
	(num_zeros, num_ones, num_agrees) = result
	print(f"Init: {i: <4} zeros = {num_zeros: <4} ones = {num_ones: <4} agrees = {num_agrees: <4}")

File: BellDriver.cs
using System;

using Microsoft.Quantum.Simulation.Core;
using Microsoft.Quantum.Simulation.Simulators;

namespace BellEntanglement {
	class BellDriver {
		static void Main(string[] args) {
			TestBellStateBase();
			TestBellStateInverse();
			TestBellStateRandom();
			TestBellState();

			System.Console.WriteLine("Press any key to continue...");
			//System.Console.ReadKey();
		}

		private static void TestBellStateBase() {
			using(var simulator = new QuantumSimulator()) {
				// Try initial values
				Result[] initials = new Result[] {Result.Zero, Result.One};

				foreach (Result initial in initials) {
					var result = BellEntanglement.TestBellStateBase.Run(simulator, 1000, initial).Result;
					var (numZeros, numOnes) = result;
					System.Console.WriteLine($"Init:{initial, -4} zeros = {numZeros, -4} ones = {numOnes, -4}");
				}
			}
		}

		private static void TestBellStateInverse() {
			using(var simulator = new QuantumSimulator()) {
				// Try initial values
				Result[] initials = new Result[] {Result.Zero, Result.One};

				foreach (Result initial in initials) {
					var result = BellEntanglement.TestBellStateInverse.Run(simulator, 1000, initial).Result;
					var (numZeros, numOnes) = result;
					System.Console.WriteLine($"Init:{initial, -4} zeros = {numZeros, -4} ones = {numOnes, -4}");
				}
			}
		}

		private static void TestBellStateRandom() {
			using(var simulator = new QuantumSimulator()) {
				// Try initial values
				Result[] initials = new Result[] {Result.Zero, Result.One};

				foreach (Result initial in initials) {
					var result = BellEntanglement.TestBellStateRandom.Run(simulator, 1000, initial).Result;
					var (numZeros, numOnes) = result;
					System.Console.WriteLine($"Init:{initial, -4} zeros = {numZeros, -4} ones = {numOnes, -4}");
				}
			}
		}

		private static void TestBellState() {
			using(var simulator = new QuantumSimulator()) {
				// Try initial values
				Result[] initials = new Result[] {Result.Zero, Result.One};

				foreach (Result initial in initials) {
					var result = BellEntanglement.TestBellState.Run(simulator, 1000, initial).Result;
					var (numZeros, numOnes, numAgrees) = result;
					System.Console.WriteLine($"Init:{initial, -4} zeros = {numZeros, -4} ones = {numOnes, -4} agrees = {numAgrees, -4}");
				}
			}
		}
	}
}